core/src/main/scala-3.0.0-RC3/cats/derived/show.scala

```diff
@@ -0,0 +1,31 @@
+package cats.derived
+
+import cats.Show
+import shapeless3.deriving.{Continue, K0, Labelling}
+
+object show extends ShowDerivation
+
+trait ShowDerivation:
+
+  extension (F: Show.type)
+    inline def derived[A](using gen: K0.Generic[A]): Show[A] = gen.derive(productShow, coproductShow)
+
+  given productShow[A](using inst : => K0.ProductInstances[Show, A], l: Labelling[A]): Show[A] with
+    def show(a: A) = {
```

Comment:
Let's use a SAM function instead (it doesn't generate anonymous classes)

Reply:
You mean define an explicit `ProductShow` trait like we've done for the other typeclasses?

Reply:
I mean you can just use a function:
`given ... : Show[A] = a => ...`